src/model/liverange.js

```diff
@@ -152,14 +152,29 @@ function transform( changeType, deltaType, batch, targetRange, sourcePosition )
 
 	const updated = Range.createFromRanges( result );
 
+	const boundariesChanged = !updated.isEqual( this );
+
+	const rangeExpanded = this.containsPosition( targetPosition );
+	const rangeShrunk = sourcePosition && ( this.containsPosition( sourcePosition ) || this.start.isEqual( sourcePosition ) );
+	const contentChanged = rangeExpanded || rangeShrunk;
+
 	// If anything changed, update the range and fire an event.
```

Comment:
This comment is outdated now, you do not update the range if content changes. But see the comment below.